melodies_monet/driver.py

```diff
@@ -1066,16 +1139,48 @@ def stats(self):
                         # Drop NaNs for model and observations in all cases.
                         pairdf = pairdf_all.reset_index().dropna(subset=[modvar, obsvar])
 
+                        if cal_reg:
+                            # Process regulatory values
+                            df1 = pairdf.copy()
+                            df1.index = df1.time_local
+                            df2 = df1.groupby("siteid").resample('H').mean().reset_index()
+
+                            if obsvar == 'PM2.5':
+                                pairdf_reg = splots.make_24hr_regulatory(df2,[obsvar,modvar]).rename(index=str,columns={obsvar+'_y':obsvar+'_reg',modvar+'_y':modvar+'_reg'})
+                            elif obsvar == 'OZONE':
+                                pairdf_reg = splots.make_8hr_regulatory(df2,[obsvar,modvar]).rename(index=str,columns={obsvar+'_y':obsvar+'_reg',modvar+'_y':modvar+'_reg'})
+                            else:
+                                print('Warning: no regulatory caculations found for ' + obsvar)
+
+                            if len(pairdf_reg[obsvar+'_reg']) == 0:
+                                print('No valid data for '+obsvar+'_reg')
+                                cal_reg = False
+                            else:
+                                # Drop NaNs for model and observations in all cases.
+                                pairdf2 = pairdf_reg.reset_index().dropna(subset=[modvar+'_reg', obsvar+'_reg'])
```

Comment:
`reset_index()` after `dropna()` instead?

Reply:
I followed the previous statement:

e.g., pairdf = pairdf_all.reset_index().dropna(subset=[modvar, obsvar])

just to make it consistent...